### PR TITLE
chore(lib/grandpa, dot/network): send `CommitMessage` directly to peer on round mismatch; cleanup grandpa `receiveMessages` 

### PR DESCRIPTION
**Pull request by noot**
## Changes

- rename `network.Service.SendMessage` -> `GossipMessage`
- create `network.Service.SendMessage` to send message to specific peer
- update grandpa vote message validation function to send `CommitMessage` (if peer's round is lower than ours) directly to peer instead of gossiping
- update grandpa `receiveMessages` function, now just start goroutine at start of round and cancel goroutine when round ends. previously was spawning multiple `receiveMessages` goroutines that were not getting shut down properly
- update genesis runtime to have c=(1,2) instead of c=(1,4) to increase block production frequency

## Tests

```
go test ./lib/grandpa -short
go test ./dot/network -short
```

## Issues

- related to #592

### CODE REVIEW

**Review by timwu20 — APPROVED** [review]

File: lib/grandpa/vote_message.go [hunk 1]
```diff
@@ -20,58 +20,49 @@ import (
...
-					"round", vm.Round,
-					"subround", vm.Message.Stage,
-					"prevote count", s.lenVotes(prevote),
-					"precommit count", s.lenVotes(precommit),
-					"votes needed", s.state.threshold(),
-				)
-			case <-ctx.Done():
-				logger.Trace("returning from receiveMessages")
-				return
+func (s *Service) receiveMessages(ctx context.Context /*cond func() bool*/) {
```
Comment (EclesioMeloJunior) [inline lib/grandpa/vote_message.go:37]: Could remove the commented code?

**Review by jimjbrettj — APPROVED** [review]

File: lib/grandpa/vote_message.go [hunk 2]
```diff
@@ -20,58 +20,49 @@ import (
...
+	for {
+		select {
+		case msg := <-s.in:
```
Comment (EclesioMeloJunior) [inline lib/grandpa/vote_message.go:40]: Could add a check to the closed channel? 
```go
case msg, ok := <-s.in:
  if !ok { ... }
```

**Review by EclesioMeloJunior — CHANGES_REQUESTED** [review]

File: lib/grandpa/vote_message.go (hunk 1 above)
Comment (noot) [reply lib/grandpa/vote_message.go:37]: done!

File: lib/grandpa/vote_message.go (hunk 2 above)
Comment (noot) [reply lib/grandpa/vote_message.go:40]: done!

**Review by EclesioMeloJunior — APPROVED** [review]

**Pull request merged**